vm/systemSmartContracts/delegation.go

```diff
@@ -1426,7 +1421,7 @@ func (d *delegation) computeAndUpdateRewards(callerAddress []byte, delegator *De
 		}
 
 		var rewardsForOwner *big.Int
-		percentage := float64(rewardData.ServiceFee) / float64(percentageDenominator)
+		percentage := float64(rewardData.ServiceFee) / float64(d.maxServiceFee)
```

Comment:
I think here we should add a check if percentage will be greater than 1. In this case rewardsForOwner will be greater than RewardToDistribute and than rewardForDelegator will be negative and so on. Something will definitely crash afterward.

Reply:
ok. will add check on constructor.

Reply:
Checked that through the code. Before saving the rewardData.ServiceFee there are strict checks. Only if we change the code in a buggy way we might have erroneous data